src/gravel/controllers/nodes/systemdisk.py

```diff
@@ -0,0 +1,255 @@
+# project aquarium's backend
+# Copyright (C) 2021 SUSE, LLC.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+import shlex
+from typing import (
+    Dict,
+    Optional,
+    List,
+)
+from logging import Logger
+from pathlib import Path
+from fastapi.logger import logger as fastapi_logger
+from pydantic.main import BaseModel
+
+from gravel.cephadm.models import (
+    NodeInfoModel,
+    VolumeDeviceModel
+)
+from gravel.controllers.errors import GravelError
+from gravel.controllers.gstate import GlobalState
+from gravel.controllers.utils import aqr_run_cmd
+
+
+logger: Logger = fastapi_logger
+
+
+class UnknownDeviceError(GravelError):
+    pass
+
+
+class UnavailableDeviceError(GravelError):
+    pass
+
+
+class MountError(GravelError):
+    pass
+
+
+class LVMError(GravelError):
+    pass
+
+
+class SystemDiskNotMountedError(GravelError):
+    pass
+
+
+class OverlayError(GravelError):
+    pass
+
+
+class MountEntry(BaseModel):
+    source: str
+    dest: str
+
+
+def get_mounts() -> List[MountEntry]:
+    proc: Path = Path("/proc/mounts")
+    assert proc.exists()
+
+    lst: List[MountEntry] = []
+    with proc.open(mode="r", encoding="utf-8") as f:
+        for line in f.readlines():
+            fields: List[str] = line.split(" ")
+            if len(fields) < 2:
+                continue
+            lst.append(MountEntry(source=fields[0], dest=fields[1]))
+    return lst
+
+
+class SystemDisk:
+
+    _gstate: GlobalState
+    _overlaydirs: Dict[str, str] = {
+        "etc": "/etc",
```

Comment:
(nit) Could just use a list and then `os.path.basenam()`

Reply:
Yeah, that's true, but I found this gives more visibility into what's happening, and the impact is negligible.